src/games/strategy/triplea/ui/export/ScreenshotExporter.java

```diff
@@ -0,0 +1,205 @@
+   * at the specified history step to that file.
+   * 
+   * @param frame The frame associated with the game screenshot to export; must not be {@code null}.
+   * @param gameData The game data; must not be {@code null}.
+   * @param node The history step at which the game screenshot is to be captured; must not be {@code null}.
+   * 
+   * @return A promise that is fulfilled when the save screenshot operation is complete or empty if the user aborted the
+   *         save; never {@code null}.
+   */
+  public static Optional<CompletableFuture<Void>> promptForFileAndRunSave(final TripleAFrame frame,
+      final GameData gameData, final HistoryNode node) {
+    final ScreenshotExporter exporter = new ScreenshotExporter(frame);
+    return exporter.promptForFile()
+        .map(file -> Optional.of(exporter.runSave(gameData, node, file)))
+        .orElse(Optional.empty());
+  }
+
+  /**
+   * Prompts the user for the file to which the screenshot will be saved.
+   * 
+   * @return The file to which the screenshot will be saved or empty if the user aborted the save; never {@code null}.
+   */
+  public Optional<File> promptForFile() {
+    final FileFilter pngFilter = new FileFilter() {
+      @Override
+      public boolean accept(final File f) {
+        if (f.isDirectory()) {
+          return true;
+        } else {
+          return f.getName().endsWith(".png");
+        }
+      }
+
+      @Override
+      public String getDescription() {
+        return "Saved Screenshots, *.png";
+      }
+    };
+    final JFileChooser fileChooser = new SaveGameFileChooser();
+    fileChooser.setFileFilter(pngFilter);
+    final int rVal = fileChooser.showSaveDialog(frame);
+    if (rVal != JFileChooser.APPROVE_OPTION) {
+      return Optional.empty();
+    }
+
+    File f = fileChooser.getSelectedFile();
+    if (!f.getName().toLowerCase().endsWith(".png")) {
+      f = new File(f.getParent(), f.getName() + ".png");
+    }
+
+    if (f.exists()) {
+      final int choice =
+          JOptionPane.showConfirmDialog(frame, "A file by that name already exists. Do you wish to over write it?",
+              "Over-write?", JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE);
+      if (choice != JOptionPane.OK_OPTION) {
+        return Optional.empty();
+      }
+    }
+
+    return Optional.of(f);
+  }
+
+  /**
+   * Saves the screenshot for the specified game at the specified history step to the specified file.
+   * 
+   * @param gameData The game data; must not be {@code null}.
+   * @param node The history step at which the game screenshot is to be captured; must not be {@code null}.
+   * @param file The file that will receive the screenshot; must not be {@code null}.
+   * 
+   * @return A promise that is fulfilled when the save screenshot operation is complete; never {@code null}.
+   */
+  public CompletableFuture<Void> runSave(final GameData gameData, final HistoryNode node, final File file) {
```

Comment:
Agreed -- another YAGNI anticipating someone would want to call `runSave()` (and `promptForFile()`) independently.